namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -76,40 +77,19 @@ private[consul] object SvcAddr {
             )
             stopped = true
             Future.Unit
-          case Throw(Failure(Some(err: ConnectionFailedException))) =>
-            log.warning(
-              "consul datacenter '%s' service '%s' retrying on connection" +
-                " failure: %s",
-              datacenter, key.name, err
-            )
-            // Drop the index, in case it's been reset by a consul restart
-            loop(None)
+
           case Throw(e) =>
-            // an error occurred. if we've previously seen a good state, fall
-            // back to that state; otherwise, fail.
+            // do not update state, log error and continue polling with backoff
             stats.errors.incr()
-            lastGood match {
-              case Some(addr) =>
-                // if we've already seen a good state, fall back to that and
-                // try again.
-                state() = addr
-                log.warning(
-                  "consul datacenter '%s' service '%s' observation error %s," +
-                    " falling back to last good state",
-                  datacenter, key.name, e
-                )
-                loop(index0)
-              case None =>
-                // if no previous good state was seen, treat the exception
-                // as effectively fatal to the service observation.
-                state() = Addr.Failed(e)
-                log.error(
-                  "consul datacenter '%s' service '%s' observation error %s," +
-                    " no previous good state to fall back to!",
-                  datacenter, key.name, e
-                )
-                Future.exception(e)
-            }
+            log.log(
+              failureLogLevel,
+              "consul datacenter '%s' service '%s' observation error %s." +
+                " Last known state is %s",
+              datacenter, key.name, e, lastGood
```

Comment:
Is the `lastGood` variable still necessary, or can we just use the current state of the activity for logging the last good state?

Reply:
In general I agree, that storing same state in 2 places isn't very pretty.

However I don't see how can we get rid of `lastGood` if we want to log last state. We handle errors inside the loop and there we only have access to `Updatable`. Even conceptually it's tricky: closure with this loop is an argument to `Var`'s "constructor", so to get current value of `Var` one would need to populate a reference to yet unconstructed `Var` to the closure that is used to construct the `Var` instance.

We still can omit the state while logging. Imo it may be usefull during troubleshooting if any sort of outage happens, but it's something super important.

Reply:
I definitely agree that logging the state is likely to be useful, and I think you're right that there's no easy way to avoid having the `lastGood` variable --- just wanted to make sure there wasn't an obvious way to remove it.